test/python/primitives/test_backend_estimator_dev.py

```diff
@@ -0,0 +1,924 @@
+        self.assertIs(call_backend, backend)
+        transpile_options = {**estimator.transpile_options.__dict__}
+        transpile_options.update({"initial_layout": expected_metadata.get("final_layout")})
+        self.assertEqual(call_kwargs, transpile_options)
+        self.assertIsInstance(composition, QuantumCircuit)
+        self.assertEqual(composition, expected_composition)
+        self.assertEqual(composition.metadata, expected_metadata)
+
+
+@ddt
+class TestComputation(TestCase):
+    """Test calculation logic."""
+
+    @data(
+        [{"0": 100, "1": 0}, "I", (1, 0)],
+        [{"0": 0, "1": 100}, "I", (1, 0)],
+        [{"0": 50, "1": 50}, "I", (1, 0)],
+        [{"0": 50, "1": 50}, "X", (0, 1)],
+        [{"0": 50, "1": 50}, "Y", (0, 1)],
+        [{"0": 50, "1": 50}, "Z", (0, 1)],
+        [{"0": 100, "1": 0}, "Z", (1, 0)],
+        [{"0": 0, "1": 100}, "Z", (-1, 0)],
+        [{"0": 80, "1": 20}, "Z", (0.6, 0.64)],
+        [{"0": 60, "1": 40}, "Z", (0.2, 0.96)],
+        [{"0": 40, "1": 60}, "Z", (-0.2, 0.96)],
+        [{"0": 20, "1": 80}, "Z", (-0.6, 0.64)],
+        [{"00": 80, "11": 20}, "ZZ", (1, 0)],
+        [{"00": 80, "10": 20}, "ZZ", (0.6, 0.64)],
+        [{"00": 20, "10": 80}, "ZZ", (-0.6, 0.64)],
+        [{"11": 80, "01": 20}, "ZZ", (0.6, 0.64)],
+        [{"11": 20, "01": 80}, "ZZ", (-0.6, 0.64)],
+        [{"00": 80, "11": 20}, "ZI", (0.6, 0.64)],
+        [{"00": 80, "10": 20}, "ZI", (0.6, 0.64)],
+        [{"00": 20, "10": 80}, "ZI", (-0.6, 0.64)],
+        [{"11": 80, "01": 20}, "ZI", (-0.6, 0.64)],
+        [{"11": 20, "01": 80}, "ZI", (0.6, 0.64)],
+        [{"11": 20, "01": 80}, "II", (1, 0)],
+    )
+    @unpack
+    def test_compute_expval_variance_pair(self, counts, pauli, expected):
+        """Test expval-variance pairs."""
+        counts = Counts(counts)
+        pauli = Pauli(pauli)
+        pair = BackendEstimator._compute_expval_variance_pair(counts, pauli)
+        self.assertEqual(pair, expected)
+
+    @data(
+        ["II", "00", +1],
+        ["II", "01", +1],
+        ["II", "10", +1],
+        ["II", "11", +1],
+        ["IX", "00", +1],
+        ["IX", "01", -1],
+        ["IX", "10", +1],
+        ["IX", "11", -1],
+        ["XI", "00", +1],
+        ["XI", "01", +1],
+        ["XI", "10", -1],
+        ["XI", "11", -1],
+        ["XX", "00", +1],
+        ["XX", "01", -1],
+        ["XX", "10", -1],
+        ["XX", "11", +1],
+        ["YZ", "00", +1],
+        ["XY", "01", -1],
+        ["ZX", "10", -1],
+        ["ZZ", "11", +1],
+        ["IXYZ", "0010", -1],
+        ["IXYZ", "1000", +1],
+        ["IXYZ", "1100", -1],
+        ["IXYZ", "1101", +1],
+        ["IXYZ", "0101", +1],
+    )
+    @unpack
+    def test_measurement_coefficient(self, pauli, bitstring, expected):
+        """Test measurement coefficients."""
+        pauli = Pauli(pauli)
+        coeff = BackendEstimator._measurement_coefficient(bitstring, pauli)
+        self.assertEqual(coeff, expected)
```

Comment:
Here's another case where a private method is being unit-tested (there's lots more in this file).  This sort of thing really tightly constrains the structure of the underlying code, whereas what we really _need_ is for all publicly visible behaviour to remain the same.  It should not be a problem if a later implementation wanted to inline this function, or move it off the class, or move it to Rust, or have it return some different object, as long as the public behaviour of the entry points remained the same.